### PR TITLE
[JIT] Add special cases for batch_norm, instance_norm in alias_analysis

### LINKED ISSUE #65563: [JIT] Fix batchnorm / instance norm alias annotations
## 🐛 Bug: mutation to batchnorm weights not reflect in alias analysis

As evidence by the Failing Test [here](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L7947), the alias annotations for batch norm and instance norm are incorrect, because the weight and bias are written to in the operator if `train=True`. We should fix the failing test and improve aliasing handling of the mutation behavior. I would start by updating the [schema](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L749) to reflect the writes. However, we want to preserve the fact that the weights are not written to in inference. I would add special handling of `aten::batchnorm` and `aten::instancenorm` so that we do not register writes either when `train` is False or `is_frozen` is true. Alias Analysis is instantiated with `is_frozen` when we apply optimizations for freezing, and we want to reflect that we'll be in inference mode there even before the `train` parameter is made to be constant. 

You can add special handling [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir/alias_analysis.cpp#L605) and read about the aliasing schema language [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/OVERVIEW.md#handling-mutability).  

You can run the test with `python test/test_ops.py -k test_variant_consistency_jit_nn_functional_batchnorm`.

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66554

Summary:

In native_functions.yaml, the schemas for batch_norm and instance_norm
are incorrect: the inputs `running_mean` and `running_var` are mutated,
but are not marked as such in the function schema. Since `(a!)?`
annotations are currently not working (see #65760), this instead adds a
special case to `alias_anaysis.cpp`. If the value of `training` or
`use_input_stats` is known to be `false`, then `alias_analysis` will
mark the input as _not_ being written to.

Test Plan:

Removed the `skip` annotation on the following test, and added a special
exception in `check_alias_annotations`:
```
python test/test_ops.py -k test_variant_consistency_jit_nn_functional_batch_norm
```

Also:
```
./build/bin/test_jit --gtest_filter="*BatchAndInstanceNormFixture*"
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31612339](https://our.internmc.facebook.com/intern/diff/D31612339)

### CODE REVIEW

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31612339).

**Review by eellison — COMMENTED** [review]
looks good !! just one comment about instance norm behavior i think we should change - i didnt realize that the writes depended also on a parameter to the module

**Review by eellison — APPROVED** [review]
nice!! LGTM

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31612339).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31612339).

**Comment by davidberard98** [comment]
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31612339).
